manticore/native/cpu/arm.py

```diff
@@ -53,25 +53,30 @@ def instruction_implementation(cpu, *args, **kwargs):
     return abstract_instruction(instruction_implementation)
 
 
+_TYPE_MAP = {
+    cs.arm.ARM_OP_REG: 'register',
+    cs.arm.ARM_OP_MEM: 'memory',
+    cs.arm.ARM_OP_IMM: 'immediate',
+    cs.arm.ARM_OP_PIMM: 'coprocessor',
+    cs.arm.ARM_OP_CIMM: 'immediate'
+}
+
+
 class Armv7Operand(Operand):
     def __init__(self, cpu, op, **kwargs):
         super().__init__(cpu, op, **kwargs)
+        self.__type = _TYPE_MAP[self.op.type]
```

Comment:
jw, why two underscores? maybe just 1 is fine

Reply:
Mostly to be more bullet proof against future changes of `Operand` - e.g. if it ever get a `_type` attribute, having a `_type` here would overwrite it. Using `__type` prevents it.

But I guess it doesn't have a `_type` and we can do it for now.

Reply:
ah i see. either way is fine with me then!